roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+import h5py
+from lazy_ops import DatasetView
+from ...segmentationextractor import SegmentationExtractor
+from ...extraction_tools import _pixel_mask_extractor
+import os
+
+class CaimanSegmentationExtractor(SegmentationExtractor):
+    """
+    This class inherits from the SegmentationExtractor class, having all
+    its funtionality specifically applied to the dataset output from
+    the \'CNMF-E\' ROI segmentation method.
+    """
+    extractor_name = 'CaimanSegmentation'
+    installed = True  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, filepath):
+        """
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing caiman *.hdmf output file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.filepath = filepath
+        self._dataset_file = self._file_extractor_read()
+        self._roi_response = self._trace_extractor_read('F_dff')
+        self._roi_response_fluorescence = self._roi_response
+        self._roi_response_neuropil = self._trace_extractor_read('C')
+        self._roi_response_deconvolved = self._trace_extractor_read('S')
+        self._images_correlation = self._summary_image_read()
+        self._raw_movie_file_location = self._dataset_file['params']['data']['fnames'][0].decode('utf-8')
+        self._sampling_frequency = self._dataset_file['params']['data']['fr'][()]
+        self.image_masks = None
+
+    def __del__(self):
+        self._dataset_file.close()
+
+    def _file_extractor_read(self):
+        f = h5py.File(self.filepath, 'r')
+        return f
+
+    def _image_mask_sparse_read(self):
+        roi_ids = self._dataset_file['estimates']['A']['indices']
+        masks = self._dataset_file['estimates']['A']['data']
+        ids = self._dataset_file['estimates']['A']['indptr']
+        return masks, roi_ids, ids
+
+    def _trace_extractor_read(self, field):
+        extracted_signals = self._dataset_file['estimates'][field] # lazy read dataset)
+        return extracted_signals
+
+    def _summary_image_read(self):
+        if self._dataset_file['estimates'].get('Cn'):
+            summary_images_ = self._dataset_file['estimates']['Cn']
+            return np.array(summary_images_).T
+        else:
+            return None
+
+    def get_accepted_list(self):
+        accepted = self._dataset_file['estimates']['idx_components']
+        if len(accepted.shape)==0:
+            accepted = list(range(self.no_rois))
+        return accepted
+
+    def get_rejected_list(self):
+        return [a for a in range(self.no_rois) if a not in set(self.get_accepted_list())]
```

Comment:
rejected is not the same as not accepted. There should be a separate rejected list.

Reply:
yes, thanks for pointing that out, caiman does have another field that explicitly stores the rejected values. I've made the changes here https://github.com/catalystneuro/roiextractors/pull/34/commits/58ee2abc76309e4e928f271c27930756cdb8a45b
Though for other classes like extract, cnmfe and sima, there is no such info so its just the setdiff with the accepted.